aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/codegen/states/DeleteUserState.kt

```diff
@@ -54,9 +54,21 @@ internal sealed class DeleteUserState : State {
                 }
                 is DeletingUser -> {
                     when (deleteUserEvent) {
-                        is DeleteUserEvent.EventType.SignOutDeletedUser -> StateResolution(UserDeleted())
+                        is DeleteUserEvent.EventType.SignOutDeletedUser -> {
+                            val action = deleteUserActions.finishDeletingUser(true)
+                            StateResolution(DeletingUser(), listOf(action))
+                        }
                         is DeleteUserEvent.EventType.ThrowError -> {
-                            StateResolution(Error(deleteUserEvent.exception))
+                            if (deleteUserEvent.signOutUser) {
+                                val action = deleteUserActions.finishDeletingUser(true)
+                                StateResolution(DeletingUser(), listOf(action))
+                            } else {
+                                StateResolution(Error(deleteUserEvent.exception))
+                            }
+                        }
+                        is DeleteUserEvent.EventType.UserSignedOutAndDeleted -> {
+                            val action = deleteUserActions.finishDeletingUser(false)
```

Comment:
this action is necessary as this will ensure the proper events are sent throughout the Authentication state machine and the authorization state machine.